core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1922,6 +1930,40 @@ private void warmup() throws InterruptedException {
         }
     }
 
+    private void maybeVerifyShareGroupStateTopicRecordCount(String persister, int messageCount) {
+        if (!persister.equals(DEFAULT_STATE_PERSISTER)) {
+            return;
+        }
+        try {
+            TestUtils.waitForCondition(() ->
+                    !cluster.brokers().get(0).metadataCache().getAliveBrokerNodes(new ListenerName("EXTERNAL")).isEmpty(),
+                DEFAULT_MAX_WAIT_MS, 100L, () -> "cache not up yet");
+            Map<String, Object> consumerConfigs = new HashMap<>();
+            consumerConfigs.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, cluster.bootstrapServers());
+            consumerConfigs.put(ConsumerConfig.GROUP_ID_CONFIG, UUID.randomUUID().toString());
```

Comment:
I think you could use `KafkaConsumer.assign()` instead so that the test doesn't have to wait for the consumer to join the consumer group.

Reply:
This too is causing flake

Reply:
I think that implies that the test would also be flaky with subscribe instead of assign, just that it runs slightly more slowly so the risk of flakiness is lower.

Reply:
Perhaps but since the pattern is same as existing tests. This addition should not decrease reliability.

Reply:
@AndrewJSchofield It was a blunder on my part. The SGS topic is set to have 3 partitions in the cluster config and I was only assigning to partition. Will rectify.